src/quasi-newton.jl

```diff
@@ -1,31 +1,78 @@
-export QuasiNewtonModel, LBFGSModel, LSR1Model
+export QuasiNewtonModel, LBFGSModel, LSR1Model, DiagonalQNModel, SpectralGradientModel
 
 abstract type QuasiNewtonModel{T, S} <: AbstractNLPModel{T, S} end
 
-mutable struct LBFGSModel{T, S, M <: AbstractNLPModel{T, S}, Meta <: AbstractNLPModelMeta{T, S}} <:
-               QuasiNewtonModel{T, S}
+mutable struct LBFGSModel{T, S, M <: AbstractNLPModel{T, S}, Meta <: AbstractNLPModelMeta{T, S},
+    Op <: LBFGSOperator{T}} <: QuasiNewtonModel{T, S}
   meta::Meta
   model::M
-  op::LBFGSOperator
+  op::Op
```

Comment:
```suggestion
  op::LBFGSOperator{T}
```
since it is not an abstract type, what do you think?

Reply:
This is not concrete:
https://github.com/JuliaSmoothOptimizers/LinearOperators.jl/blob/9591943da03d9f113f47449447df2603305d2d74/src/lbfgs.jl#L52

Reply:
you are right, thanks